src/EnergyPlus/MixedAir.cc

```diff
@@ -1363,64 +1363,65 @@ void GetOAControllerInputs(EnergyPlusData &state)
             // Zone maximum outdoor air fraction
             thisVentilationMechanical.ZoneMaxOAFraction = NumArray(1);
 
-            state.dataMixedAir->VentMechZoneOrListName.allocate(NumGroups);
-            state.dataMixedAir->DesignSpecOAObjName.allocate(NumGroups);
-            state.dataMixedAir->DesignSpecOAObjIndex.dimension(NumGroups, 0);
-            state.dataMixedAir->DesignSpecZoneADObjName.allocate(NumGroups);
-            state.dataMixedAir->DesignSpecZoneADObjIndex.dimension(NumGroups, 0);
+            EPVector<int> designSpecOAObjIndex;           // index of the design specification outdoor air object
+            EPVector<std::string> ventMechZoneOrListName; // Zone or Zone List to apply mechanical ventilation rate
+            EPVector<int> designSpecZoneADObjIndex;       // index of the design specification zone air distribution object
+            ventMechZoneOrListName.allocate(NumGroups);
+            designSpecOAObjIndex.dimension(NumGroups, 0);
+            designSpecZoneADObjIndex.dimension(NumGroups, 0);
 
             //   First time through find the total number of zones requiring mechanical ventilation
             //   May include duplicate zones. Will check for duplicate zones further down in this subroutine.
             for (int groupNum = 1; groupNum <= NumGroups; ++groupNum) {
-                state.dataMixedAir->VentMechZoneOrListName(groupNum) = AlphArray((groupNum - 1) * 3 + 5);
+                ventMechZoneOrListName(groupNum) = AlphArray((groupNum - 1) * 3 + 5);
 
                 //     Getting OA details from design specification OA object
                 if (!lAlphaBlanks((groupNum - 1) * 3 + 6)) {
-                    state.dataMixedAir->DesignSpecOAObjName(groupNum) = AlphArray((groupNum - 1) * 3 + 6);
-                    int ObjIndex = Util::FindItemInList(state.dataMixedAir->DesignSpecOAObjName(groupNum), state.dataSize->OARequirements);
-                    state.dataMixedAir->DesignSpecOAObjIndex(groupNum) = ObjIndex;
-
-                    if (ObjIndex == 0) {
-                        ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisVentilationMechanical.Name));
-                        ShowContinueError(state,
-                                          format("... not found {}=\"{}\".",
-                                                 cAlphaFields((groupNum - 1) * 3 + 6),
-                                                 state.dataMixedAir->DesignSpecOAObjName(groupNum)));
-                        ErrorsFound = true;
+                    std::string_view designSpecOAObjName = AlphArray((groupNum - 1) * 3 + 6);
+                    if (designSpecOAObjName.empty()) {
```

Comment:
How can this string be empty when line 1397 says it's not blank?

Reply:
> How can this string be empty when line 1379 says it's not blank? I think you want to remove the outer IF block.

Well, it can't. That's what I get for blindly moving things around. Fixed.